lib/ohai/plugins/cpu.rb

```diff
@@ -344,12 +344,20 @@ def parse_cpuinfo
     cpuinfo
   end
 
+  # Check if the `lscpu` data looks reasonable
+  def valid_lscpu?(lscpu)
+    return false if lscpu.empty?
+    return false if %i{total real cores}.any? { |key| lscpu[key].to_i == 0 }
+
+    true
+  end
+
   collect_data(:linux) do
     cpuinfo = parse_cpuinfo
     lscpu = parse_lscpu(cpuinfo)
 
-    # If we don't have any data from lscpu then get it from /proc/cpuinfo
-    if lscpu.empty?
+    # If we don't have any sensible data from lscpu then get it from /proc/cpuinfo
+    if !valid_lscpu?(lscpu)
```

Comment:
Could you reverse this, so that: 

```
if valid_lscpu?(lscpu) 
  cpu lscpu
else 
...
````
Keeping the positive condition first is consistent with style throughout chef-infra.